service/gov_pay.go

```diff
@@ -141,6 +141,44 @@ func (gp *GovPayService) getGovPayPaymentState(paymentResource *models.PaymentRe
 	return &govPayResponse.State, Success, nil
 }
 
+// GetGovPayPaymentDetails gets the details of a GovPay payment
+func (gp *GovPayService) GetGovPayPaymentDetails(paymentResource *models.PaymentResourceRest) (*models.PaymentDetails, ResponseType, error) {
```

Comment:
the bulk of this function (apart from response type and last 2 lines where that is created) appears to be the same as `getGovPayPaymentState` in this file. Could we refactor our a function to avoid this duplication?

seems like both take in PaymentResourceRest and use config to decide the token to use, call gov pay and get an incoming gov pay response struct - the shared function could return that incoming gov pay response and callers can then construct the data they want from it